agent/src/agent/destination/anodot_api_client.py

```diff
@@ -35,22 +36,29 @@ class AnodotApiClientException(click.ClickException):
 
 
 class AnodotApiClient:
-    def __init__(self, access_key, proxies, base_url='https://app.anodot.com'):
-        self.access_key = access_key
-        self.base_url = base_url
-        self.proxies = proxies
+    def __init__(self, destination_: HttpDestination):
+        self.url = destination_.url
+        self.proxies = proxy.get_config(destination_.proxy)
         self.session = requests.Session()
-        self._get_auth_token()
-
-    def _get_auth_token(self):
-        auth_token = self.session.post(self._build_url('access-token'),
-                                       json={'refreshToken': self.access_key},
-                                       proxies=self.proxies)
-        auth_token.raise_for_status()
-        self.session.headers.update({'Authorization': 'Bearer ' + auth_token.text.replace('"', '')})
+        self.session.headers.update({'Authorization': 'Bearer ' + self._get_auth_token(destination_)})
```

Comment:
Need to check the token every time the request is made, not only on object creation